### PR TITLE
Calculate the mixing of school populations assuming high levels of cl…

### PR DESCRIPTION
**Pull request by anthonyohare**
Calculation of school age mixing for the mixing matrix using the number, type, and size of schools in each local authority 
Assumes a high level of within class mixing, a lower level of within school mixing and quite a low level of between school mixing. The calculated value is inserted in the full_mixing_matrix.

I have separated primary and secondary schools and combine the statistics at a local authority level, it will be quite easy to seperate this into mixing rates for [5-12) and [12-28) if needed later.

There will be some code here that may/will be used in other age group mixing so I will likely tidy this up later, splitting off the HB-LA data structures.

### CODE REVIEW

**Comment by bobturneruk** [comment]
Thanks for putting this up @anthonyohare! It will be of great interest to everyone involved with the data side of the operation (e.g. @mrow84 and @rtblackwell) as it exposes some of the challenges we're facing.

File: sample_input_files/calculateSchoolAgeMixing.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+# The total number of contacts in a local authority is an average of all the contacts wєighted according to the 
+# school size.
+# 
+# A similar weighting across local authorities in a health board is done.
+
+# The weighting factors are asssumed to be:
+within_class_mixing = 1.0
+within_school_mixing = 0.4
+within_la_mixing = 0.01
+ave_class_size = 30
```
Comment (aflag) [inline sample_input_files/calculateSchoolAgeMixing.py:199]: Are these factors calculated from some source or are they just guesses? I think modelling just within class mixing is probably good enough. I think trying to add other types of mixing will just complicate the model without bringing more certainty.

File: sample_input_files/calculateSchoolAgeMixing.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (bobturneruk) [reply sample_input_files/calculateSchoolAgeMixing.py:1]: Agree on having a new folder for it - I think it needs to be in this repository for now.

File: sample_input_files/calculateSchoolAgeMixing.py (hunk 1 above)
Comment (anthonyohare) [reply sample_input_files/calculateSchoolAgeMixing.py:199]: just guesses for now. The within_school and between school rates is really to have a weight when averaging over all schools in a local authority. If we do just within_class I would expect the rate to be ave_class_size/num_classes_in_the_school. You're right about over complicating the model (especially adding in rates that are just guesses) but just putting in within_class mixing will give a rate that is lower than reality.

Comment (anthonyohare) [reply sample_input_files/calculateSchoolAgeMixing.py:199]: A similar consideration should be made when we calculate the mixing rates for adults (within_office?, within_factory? etc)

File: sample_input_files/calculateSchoolAgeMixing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anthonyohare) [reply sample_input_files/calculateSchoolAgeMixing.py:1]: Agree with @aflag until I get my head around the data pipeline, where this ultimately belongs, I think a temporary dir in this repo is ok for now (issue-507 had created a scripts directory but I'll change it to extractors and put this in here too).

**Comment by anthonyohare** [comment]
Not required anymore; implemented elsewhere

**Pull request closed without merging**